api/envoy/config/filter/http/ext_authz/v2alpha/ext_authz.proto

```diff
@@ -48,8 +48,10 @@ message ExtAuthz {
 //
 // A failed check will cause this filter to close the HTTP request normally with 403 (Forbidden),
 // unless a different status code has been indicated by the authorization server via response
-// headers. If other headers in the authorization response need to be sent to client, this can also
-// be done by specifying them in *allowed_authorization_headers*.
+// headers. Note that if any of the 5xx HTTP error codes or an error is originated during this
```

Comment:
I think this is changing the behavior of the existing implementation if I understand the wording here. Is that correct? Earlier even if external auth service is not reachable, if failure_mode_allow is set to true we used to allow the request to take its own path. Is that changing now?

Reply:
`I think this is changing the behaviour of the existing implementation if I understand the wording here. Is that correct?`

Yes, that's right. Now, we treat any 5xx like an error. Before the client would receive a Forbidden response even if failure_mode_allow was true.

`Earlier even if external auth service is not reachable, if failure_mode_allow is set to true we used to allow the request to take its own path. Is that changing now?`

This is not changing. The only difference is that 5xx are treated in the same manner. E.g., if the authz server returns 500 and failure_mode_allow is true, the call follows the path, otherwise, it returns a 403 to the client and logs the 500.

Please, let me know if it doesn't make sense.

Reply:
@gsagula Thanks for the explanation. It completely makes sense now. I will review it again in a bit.

Reply:
No problem, thank you 👍Take your time.

Reply:
Can you document clearly the two different cases? When `failure_mode_allow` is set and when not, what the behaviors are, ideally as a bullet point list. This will make it easier to understand.

Also "error is originated" is unclear to me, you explain this much better in this thread, where you talk about this in terms of what the authz server is returning.